bindings/java/src/main/native/endpoint.cc

```diff
@@ -73,8 +80,13 @@ Java_org_openucx_jucx_ucp_UcpEndpoint_createEndpointNative(JNIEnv *env, jclass c
         ep_params.conn_request = reinterpret_cast<ucp_conn_request_h>(env->GetLongField(ucp_ep_params, field));
     }
 
-    ep_params.field_mask |= UCP_EP_PARAM_FIELD_ERR_HANDLER;
-    ep_params.err_handler.cb = error_handler;
+    if (ep_params.field_mask & UCP_EP_PARAM_FIELD_ERR_HANDLER) {
+        field = env->GetFieldID(ucp_ep_params_class, "errorHadnlerPrivate",
+                                "Lorg/openucx/jucx/ucp/UcpEndpoint$ErrorHandlerPrivate;");
```

Comment:
maybe use Pair<UcpEndpoint, UcpEndpointErrorHandler> instead of extra inner class ErrorHandlerPrivate ?